### DISCUSSION TITLE
List user passkeys

### DISCUSSION THREAD

**Discussion by JStonevalley** [MasterKale/SimpleWebAuthn#421] [General]
Hi, I have been looking through the docs and code to see if it is possible to extract more information about a passkey during registration in order to be able to list it to the user? Preferably something like device name? I know that passkeys in github and google somehow gather that information during the registration process.

**Comment by MasterKale** [comment]
@JStonevalley Unfortunately there's no consistent way to identify platform authenticators during registration. Typically the AAGUID you'd get back from a `"direct"` attestation statement would be how you'd identify the make and model of the authenticator. However platform authenticators have had historically spotty-to-nonexistent support for attestation.

This reality has had most RP's, like GitHub and Google, build out their own heuristics to look at signals including user agent, `authenticatorAttachment` in the registration response, and more recently unattested AAGUIDs from certain platform authenticators to intuit a user-friendly label for a newly-registered passkey.
